pkg/validation/validation.go

```diff
@@ -2,141 +2,9 @@ package validation
-			digit = true
-		}
-	}
-
-	if !uppercase {
-		errs = append(errs, fmt.Errorf("password must contain an uppercase character"))
-	}
-
-	if !lowercase {
-		errs = append(errs, fmt.Errorf("password must contain a lowercase character"))
-	}
-
-	if !digit {
-		errs = append(errs, fmt.Errorf("password must contain a digit"))
-	}
-
-	return
-}
-
-// ValidateIsNotAccountLocator validates that the account value is not an account locator. Account locators have the
-// following format: 8 characters where the first 3 characters are letters and the last 5 are digits. ex: ABC12345
-// The desired format should be 'organization_name.account_name' ex: testOrgName.testAccName.
-func ValidateIsNotAccountLocator(i interface{}, k string) (s []string, errors []error) {
-	v, ok := i.(string)
-	if !ok {
-		errors = append(errors, fmt.Errorf("expected type of %s to be string", k))
-		return
-	}
-	if !strings.Contains(v, ".") {
-		errors = append(errors, fmt.Errorf("account locators are not allowed - please use 'organization_name.account_name"))
-		return
-	}
-	if len(v) == 8 {
-		isAccountLocator := true
-		firstHalf := v[0:3]
-		for _, r := range firstHalf {
-			if !unicode.IsLetter(r) {
-				isAccountLocator = false
-			}
-		}
-		secondHalf := v[3:]
-		for _, r := range secondHalf {
-			if !unicode.IsDigit(r) {
-				isAccountLocator = false
-			}
-		}
-		if isAccountLocator {
-			errors = append(errors, fmt.Errorf("account locators are not allowed - please use 'organization_name.account_name"))
-		}
-	}
-	return
-}
-
-func ValidateAccountIdentifier(i interface{}, k string) (s []string, errors []error) {
-	v, ok := i.(string)
-	if !ok {
-		errors = append(errors, fmt.Errorf("expected type of %s to be string", k))
-		return
-	}
-
-	match, _ := regexp.MatchString(`^[a-zA-Z][a-zA-Z0-9_]*$`, v)
-	if !match {
-		errors = append(errors, fmt.Errorf("must start with an alphabetic character and cannot contain spaces or special characters except for underscores (_)"))
-	}
-	return
-}
-
-func ValidateEmail(i interface{}, k string) (s []string, errors []error) {
-	v, ok := i.(string)
-	if !ok {
-		errors = append(errors, fmt.Errorf("expected type of %s to be string", k))
-		return
-	}
-
-	match, _ := regexp.MatchString(`^[a-zA-Z0-9._%+\-]+@[a-zA-Z0-9.\-]+\.[a-zA-Z]{2,}$`, v)
-	if !match {
-		errors = append(errors, fmt.Errorf("must be a valid email address"))
-	}
-	return
-}
-
-// ValidateAdminName: A login name can be any string consisting of letters, numbers, and underscores. Login names are always case-insensitive.
-func ValidateAdminName(i interface{}, k string) (s []string, errors []error) {
-	v, ok := i.(string)
-	if !ok {
-		errors = append(errors, fmt.Errorf("expected type of %s to be string", k))
-		return
-	}
-
-	match, _ := regexp.MatchString(`^[a-zA-Z0-9_]+$`, v)
-	if !match {
-		errors = append(errors, fmt.Errorf("must be a valid admin name"))
-	}
-	return
-}
-
 func FormatFullyQualifiedObjectID(dbName, schemaName, objectName string) string {
```

Comment:
These functions don't validate anything, can they be moved somewhere else?
Just an observation: usage of `FormatFullyQualifiedObjectID` could be potentially replaced by `SchemaObjectIdentifier` during ID rework.

Reply:
Also, moved to other package (as suggested) and renamed the file.